From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Thu, 7 Jul 2011 18:19:04 +0200
Subject: debianize error messages

--- a/IPython/scripts/ipython
+++ b/IPython/scripts/ipython
@@ -4,4 +4,13 @@
 
 from IPython.frontend.terminal.ipapp import launch_new_instance
 
-launch_new_instance()
+try:
+  launch_new_instance()
+except ImportError as e:
+  if "qt.console.qtconsoleapp" in e.message:
+    print "Could not start qtconsole. Please install ipython-qtconsole"
+  elif "html.notebook.notebookapp" in e.message:
+    print "Could not start notebook. Please install ipython-notebook"
+  else:
+    # if there is no clue on the cause -- just re-raise
+    raise
--- a/IPython/zmq/__init__.py
+++ b/IPython/zmq/__init__.py
@@ -50,12 +50,12 @@ def check_for_zmq(minimum_version, modul
     try:
         import zmq
     except ImportError:
-        raise ImportError("%s requires pyzmq >= %s"%(module, minimum_version))
+        raise ImportError("%s requires python-zmq >= %s"%(module, minimum_version))
 
     pyzmq_version = zmq.__version__
     
     if 'dev' not in pyzmq_version and V(pyzmq_version) < V(minimum_version):
-        raise ImportError("%s requires pyzmq >= %s, but you have %s"%(
+        raise ImportError("%s requires python-zmq >= %s, but you have %s"%(
                         module, minimum_version, pyzmq_version))
 
     if V(zmq.zmq_version()) >= V('4.0.0'):
--- a/IPython/lib/clipboard.py
+++ b/IPython/lib/clipboard.py
@@ -43,8 +43,9 @@ def tkinter_clipboard_get():
     try:
         import Tkinter
     except ImportError:
+        from IPython.utils import py3compat
         raise TryNext("Getting text from the clipboard on this platform "
-                      "requires Tkinter.")
+                      "requires the %s-tk package." % ("python3" if py3compat.PY3 else "python"))
     root = Tkinter.Tk()
     root.withdraw()
     text = root.clipboard_get()
