From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Thu, 7 Jul 2011 18:19:04 +0200
Subject: debianize error messages

--- a/IPython/zmq/__init__.py
+++ b/IPython/zmq/__init__.py
@@ -50,12 +50,12 @@ def check_for_zmq(minimum_version, modul
     try:
         import zmq
     except ImportError:
-        raise ImportError("%s requires pyzmq >= %s"%(module, minimum_version))
+        raise ImportError("%s requires python-zmq >= %s"%(module, minimum_version))
 
     pyzmq_version = zmq.__version__
     
     if 'dev' not in pyzmq_version and V(pyzmq_version) < V(minimum_version):
-        raise ImportError("%s requires pyzmq >= %s, but you have %s"%(
+        raise ImportError("%s requires python-zmq >= %s, but you have %s"%(
                         module, minimum_version, pyzmq_version))
 
     if V(zmq.zmq_version()) >= V('4.0.0'):
--- a/IPython/lib/clipboard.py
+++ b/IPython/lib/clipboard.py
@@ -43,8 +43,9 @@ def tkinter_clipboard_get():
     try:
         import Tkinter
     except ImportError:
+        from IPython.utils import py3compat
         raise TryNext("Getting text from the clipboard on this platform "
-                      "requires Tkinter.")
+                      "requires the %s-tk package." % ("python3" if py3compat.PY3 else "python"))
     root = Tkinter.Tk()
     root.withdraw()
     text = root.clipboard_get()
--- a/IPython/frontend/terminal/ipapp.py
+++ b/IPython/frontend/terminal/ipapp.py
@@ -384,10 +384,19 @@ def load_default_config(ipython_dir=None
 
 def launch_new_instance():
     """Create and run a full blown IPython instance"""
-    app = TerminalIPythonApp.instance()
-    app.initialize()
-    app.start()
-
+    try:
+        app = TerminalIPythonApp.instance()
+        app.initialize()
+        app.start()
+    except ImportError as e:
+        from IPython.utils import py3compat
+        ipy = "ipython3" if py3compat.PY3 else "ipython"
+        if "IPython.frontend.qt" in str(e) or "qt.console.qtconsoleapp" in str(e):
+            print "Could not start qtconsole. Please install %s-qtconsole" % ipy
+        elif "IPython.frontend.html" in str(e)  or "html.notebook.notebookapp" in str(e):
+            print "Could not start notebook. Please install %s-notebook" % ipy
+        else:
+            raise
 
 if __name__ == '__main__':
     launch_new_instance()
