From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Thu, 7 Jul 2011 18:19:04 +0200
Subject: improve error message on missing ipython-qtconsole

---
 IPython/scripts/ipython |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/IPython/scripts/ipython b/IPython/scripts/ipython
index 20095a8..e8019e4 100755
--- a/IPython/scripts/ipython
+++ b/IPython/scripts/ipython
@@ -7,4 +7,8 @@ entry points."""
 
 from IPython.frontend.terminal.ipapp import launch_new_instance
 
-launch_new_instance()
+try:
+  launch_new_instance()
+except ImportError as e:
+  if "qt.console.qtconsoleapp" in e.message:
+    print "Could not start qtconsole. Please install ipython-qtconsole"
-- 
