Description: use packaged lessc compiler
--- a/IPython/html/fabfile.py
+++ b/IPython/html/fabfile.py
@@ -27,7 +27,7 @@ def _compile_less(source, target, minify
     verbose = _to_bool(verbose)
     min_flag = '-x' if minify is True else ''
     ver_flag = '--verbose' if verbose is True else ''
-    lessc = os.path.join('components', 'less.js', 'bin', 'lessc')
+    lessc = 'lessc'
     with lcd(static_dir):
         local('{lessc} {min_flag} {ver_flag} {source} {target}'.format(**locals()))
 
