Description: default debian backend is tk
Forwarded: not-needed
--- a/IPython/core/tests/test_pylabtools.py
+++ b/IPython/core/tests/test_pylabtools.py
@@ -196,24 +196,24 @@ class TestPylabSwitch(object):
     def test_qt(self):
         s = self.Shell()
         gui, backend = s.enable_matplotlib(None)
-        nt.assert_equal(gui, 'qt')
-        nt.assert_equal(s.pylab_gui_select, 'qt')
+        nt.assert_equal(gui, 'tk')
+        nt.assert_equal(s.pylab_gui_select, 'tk')
 
         gui, backend = s.enable_matplotlib('inline')
         nt.assert_equal(gui, 'inline')
-        nt.assert_equal(s.pylab_gui_select, 'qt')
+        nt.assert_equal(s.pylab_gui_select, 'tk')
 
-        gui, backend = s.enable_matplotlib('qt')
-        nt.assert_equal(gui, 'qt')
-        nt.assert_equal(s.pylab_gui_select, 'qt')
+        gui, backend = s.enable_matplotlib('tk')
+        nt.assert_equal(gui, 'tk')
+        nt.assert_equal(s.pylab_gui_select, 'tk')
 
         gui, backend = s.enable_matplotlib('inline')
         nt.assert_equal(gui, 'inline')
-        nt.assert_equal(s.pylab_gui_select, 'qt')
+        nt.assert_equal(s.pylab_gui_select, 'tk')
 
         gui, backend = s.enable_matplotlib()
-        nt.assert_equal(gui, 'qt')
-        nt.assert_equal(s.pylab_gui_select, 'qt')
+        nt.assert_equal(gui, 'tk')
+        nt.assert_equal(s.pylab_gui_select, 'tk')
 
     def test_inline(self):
         s = self.Shell()
