From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Sun, 18 Dec 2011 14:04:29 +0100
Subject: use system mathjax if available

--- a/IPython/html/notebookapp.py
+++ b/IPython/html/notebookapp.py
@@ -217,6 +217,11 @@ class NotebookWebApplication(web.Applica
             pattern = url_path_join(settings['base_url'], handler[0])
             new_handler = tuple([pattern] + list(handler[1:]))
             new_handlers.append(new_handler)
+
+        if os.path.exists("/usr/share/javascript/mathjax/MathJax.js"):
+            new_handlers.append((r"/mathjax/(.*)", web.StaticFileHandler,
+                                {"path": "/usr/share/javascript/mathjax"}))
+
         # add 404 on the end, which will catch everything that falls through
         new_handlers.append((r'(.*)', Template404))
         return new_handlers
@@ -487,6 +492,10 @@ class NotebookApp(BaseIPythonApplication
                 url = url_path_join(url_prefix, u"mathjax/MathJax.js")
                 self.log.info("Serving local MathJax from %s at %s", mathjax, url)
                 return url
+
+        if os.path.exists("/usr/share/javascript/mathjax/MathJax.js"):
+            self.log.info("Using system MathJax")
+            return u"/mathjax/MathJax.js"
         
         # no local mathjax, serve from CDN
         if self.certfile:
